src/Algebra/Graph.hs

```diff
@@ -176,7 +178,15 @@ data Graph a = Empty
              | Vertex a
              | Overlay (Graph a) (Graph a)
              | Connect (Graph a) (Graph a)
-             deriving (Functor, Show)
+             deriving (Show)
+
+instance Functor Graph where
+  fmap = mapG
```

Comment:
Should this also be called `mapR` to highlight that it is there only for rewrite rules?

Reply:
See the note I've added above. I think it is just simpler to choose one consistent naming convention and follow it throughout the code. So, let's add `R` everywhere: `eqR`, `eqIntR`, etc. 

> and more prosaically I already have a `mapGR` used

Let's not invent new names. The above, for example, should be `fmapR` (it's a substitute for `fmap`). And the function you call `mapGR` should probably be more appropriately called `composeR`:

```
composeR :: (b -> c) -> (a -> b) -> a -> c
composeR = (.)
```

I find `mapGR = (.)` to be quite confusing.

Reply:
Ok, indeed it can be more simple to understand for anyone reading the code